ecs/pod_creator_test.go

```diff
@@ -1,12 +1,286 @@
+			podCreator, err := NewBasicECSPodCreator(nil, nil)
+			require.Error(t, err)
+			require.Zero(t, podCreator)
+		},
+		"CreatePodFailsWithInvalidCreationOpts": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			opts := cocoa.NewECSPodCreationOptions()
+			require.NotZero(t, opts)
+			assert.Zero(t, *opts)
+
+			podCreator, err := NewBasicECSPodCreator(c, nil)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithSecretsNoVault": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("evName").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName("name").SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar})
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().SetName("name").
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole("role").
+				AddTags("tag").
+				SetExecutionOptions(*execOpts)
+			require.NotZero(t, opts)
+			assert.NotZero(t, *opts)
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, nil)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithIncompleteContainerDef": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
+			envVar := cocoa.NewEnvironmentVariable().SetName("ev").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name")).SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar}).
+				SetName("container")
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().SetName("name").
+				AddContainerDefinitions(*containerDef).
+				SetMemoryMB(128).
+				SetCPU(128).
+				SetTaskRole("role").
+				AddTags("tag").
+				SetExecutionOptions(*execOpts)
+			require.NotZero(t, opts)
+			assert.NotZero(t, *opts)
+			assert.NoError(t, opts.Validate())
+
+			podCreator, err := NewBasicECSPodCreator(c, v)
+			require.NoError(t, err)
+
+			p, err := podCreator.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithInvalidSecretName": func(ctx context.Context, t *testing.T, c cocoa.ECSClient, v cocoa.Vault) {
```

Comment:
What exactly is invalid about the secret name? I'm pretty sure the options validation in the pod creator would prevent it from being an invalid name. Do you mean to actually test the condition "the secret doesn't exist in Secrets Manager, but we also don't create it"?

Reply:
I don't recall the point of this test, so I'm just going to remove it. If the secret doesn't exist in SM and we don't create it, is that when we declare a secret and its fields but the container definition doesn't include it in its environment variable list?

Reply:
Removing the test SGTM.